.github/workflows/deploy.yml

```diff
@@ -0,0 +1,98 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - 'DOI-887_*'
+
+jobs:
+  cancel-previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: cancel-previous
+    env:
+      SB_APP_DIR: $HOME/work/switchboad-dapp/switchboad-dapp
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '10.22.0'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Get branch
+        uses: tj-actions/branch-names@v2.1
+        id: branch-name
+
+      - name: Get commit hash
+        id: commithash
+        if: github.event_name == 'push'
+        run: |
+          echo ${{ steps.branch-name.outputs.current_branch }}
+          echo $GITHUB_SHA | cut -c1-8
+
+      - name: Setup AWS Host machine
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ca-central-1
+
+      - name: Install npm packages
+        run: |
+          npm install
+          envsubst < src/environments/constants.template.ts > src/environments/constants.ts
+
+      - name: Prepare for master branch
+        if: github.ref == 'refs/heads/master'
+        run: |
+          cp src/environments/environment.prod.ts src/environments/environment.ts
+          echo "Prod Env Enabled"
+          npm run build-prod
+
+      - name: Prepare for non-master branch
+        if: github.ref != 'refs/heads/master'
+        run: |
+          cp src/environments/environment.dev.ts src/environments/environment.ts
+          echo "Prod Env Disabled"
+          npm run build
+
+      - name: Checkout develop branch
+        uses: actions/checkout@v2
+        with:
+          ref: develop
+
+      - name: Run Test
+        run: |
+          npm install
+          npm run test:headless
+
+      - name: Deploy on develop branch
+        if: github.ref == 'refs/heads/DOI-887_*'
+        run: |
+          rsync -e "ssh -o StrictHostKeyChecking=no" -r dist/* ${{ secrets.SB_WEB_HOST }}:${{ env.SB_APP_DIR }}/develop
```

Comment:
Why the checkout of the develop branch? Is it just while you were testing your CI changes?

Reply:
Already fixed 👍